### PR TITLE
Feature: add `apko_config` data source.

### PR DESCRIPTION
**Pull request by mattmoor**
:gift: This adds an `apko_config` data source for accessing the configuration information in a structured way (e.g. the architectures to build for)

/kind feature

### CODE REVIEW

**Comment by mattmoor** [comment]
@imjasonh with this, I'm able to make my little publisher module work.  Wanted to get a sniff test on this before expanding this to include any more of the schema.

We should also discuss if/what other parts of the schema we would want to implement for an MVP, since it is pretty tedious to write this stuff (LMK if you see anything we could be doing better).

**Comment by mattmoor** [comment]
Example: 
image

From:
```hcl
resource "apko_build" "this" {
  repo   = var.target_repository
  config = var.config
}

data "apko_config" "this" {
  config = var.config
}

resource "cosign_attest" "sboms" {
  for_each = toset(concat(data.apko_config.this.data.archs, ["index"]))

  image          = apko_build.this.sboms[each.key].digest
  predicate_type = "https://spdx.dev/Document"
  predicate      = apko_build.this.sboms[each.key].sbom
}
```

**Comment by mattmoor** [comment]
Example usage: https://github.com/mattmoor/terraform-publisher-apko/blob/8cae6384218e90a2d9be475bf74fb3fdfd622bf0/main.tf#L26-L36

**Comment by imjasonh** [comment]
(not blocking this PR, just noticed this before diving into the diff) Re: your example, can you make `predicate_type` based on the `apko_build.sbom` attribute? Should that have some `.sbom.predicate_type` attribute so that we can plumb it through directly?

File: internal/provider/build.go
Comment (imjasonh) [inline internal/provider/build.go:50]: I'd either add a comment to these, or better, change `sbom string` to `content []byte` so it's clearer what kind of data that's going to be.

File: internal/provider/config_data_source.go
Comment (imjasonh) [inline internal/provider/config_data_source.go:40]: Is the idea that eventually this type could also be used in `BuildResourceModel`, so that folks can put the whole (or at least parts) of the config directly in HCL?

I could see a provider-wide `repositories` and `keyring` that get inherited by all `apko_build`s for example, or a `variable "packages"` that gets piped into `apko_build.foo.config.packages`.

File: internal/provider/config_data_source.go
Comment (imjasonh) [inline internal/provider/config_data_source.go:53]: Maybe `config_file`? `config_contents`?

So that `data` can become `config` or `parsed_config` or something more descriptive?

File: internal/provider/config_data_source.go
Comment (mattmoor) [reply internal/provider/config_data_source.go:40]: Yeah, I don't want to block on that, but I think there are a lot of fun things we could do in that direction once we unblock attestations.

**Comment by mattmoor** [comment]
Ref: https://github.com/chainguard-dev/terraform-provider-apko/issues/45

File: internal/provider/build.go
Comment (mattmoor) [reply internal/provider/build.go:50]: I renamed it to `predicate` and added a `predicate_type` field.

File: internal/provider/config_data_source.go
Comment (mattmoor) [reply internal/provider/config_data_source.go:53]: I'm going with `config_contents` in and `config` out

**Comment by mattmoor** [comment]
New version of the config:
```hcl
data "apko_config" "this" {
  config_contents = var.config
}

resource "cosign_attest" "sboms" {
  for_each = toset(concat(data.apko_config.this.config.archs, ["index"]))

  image          = apko_build.this.sboms[each.key].digest
  predicate_type = apko_build.this.sboms[each.key].predicate_type
  predicate      = apko_build.this.sboms[each.key].predicate
}
```

File: internal/provider/resource_build.go
Comment (mattmoor) [inline internal/provider/resource_build.go:40]: I'd love to replace this with a structured type.  LMK if you have ideas @imjasonh because I've hit problems trying to use a structured type here.

**Review by imjasonh — APPROVED** [review]

**Pull request merged**
